### DISCUSSION TITLE
How do i read a multipart/related content

### DISCUSSION THREAD

**Discussion by bhuvikris** [reactiveui/refit#1571] [Q&A] [answered]
Using refit, how can i read a Multipart response. It is a Get call.
Accept: multipart/related
How do deal with partial content response ?

**Comment by glennawatson** [comment, marked as answer]
Refit does not parse `multipart/related` responses for you. There is no built-in multipart-response reader; built-in raw types on the response side are `Stream`, `string`, `HttpResponseMessage` and `HttpContent`.

Approach:
- Declare the method to return the raw response and parse the parts yourself:

```csharp
[Get("/resource")]
[Headers("Accept: multipart/related")]
Task<HttpResponseMessage> GetParts();
```

Then read it with `MultipartMemoryStreamProvider` from `System.Net.Http`:

```csharp
using var resp = await api.GetParts();
var multipart = await resp.Content.ReadAsMultipartAsync();
foreach (var part in multipart.Contents)
{
    var bytes = await part.ReadAsByteArrayAsync();
    // inspect part.Headers for content-type / content-id
}
```

Notes:
- `ReadAsMultipartAsync` lives in the `System.Net.Http.Formatting` package (Microsoft.AspNet.WebApi.Client). If you prefer no extra dependency, read the raw `Stream` and split on the boundary from the `Content-Type` header yourself.
- For partial content (HTTP 206 / `Range`), set the `Range` header (e.g. via `[Headers]` or a `HttpRequestMessage`) and read `resp.StatusCode` / `Content-Range` from the returned `HttpResponseMessage`.

In short: return `HttpResponseMessage` from Refit and do the multipart/range handling with the standard HttpClient types.
